openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
-            var catData = [];
-
-            for (var k in scaledCatIndicator) {
-                catData.push(scaledCatIndicator[k]);
-            }
-
-            // TODO remove this concat if need be
-            var concat = catData.concat(previousCatData);
-
-            Category_PCP_Chart(catData, municipality, districName, concat);
-
-            for (var k in scaledCatIndicator) {
-                previousCatData.push(scaledCatIndicator[k]);
-            }
-
-            /////////////////////////////////////////////
-            /////////// Create the svi object ///////////
-            /////////////////////////////////////////////
-
-            for (var k in sviIndicator) {
-                delete sviIndicator[k];
-            }
-
-            for (var k in scaledSviIndicator) {
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
+                var munic4 = la[o].properties[selectedRegion];
+                var theme4 = name;
+                tempString.push(munic4 + '|'+ theme4 +'|'+ tempSum);
             }
+        }
+    }
 
-            // Associate the newSVIvalues method with the svi object
-            sviIndicator.SVIvalues = newSVIvalues;
-            var tempSviParentChildKey = [];
-            var tempSviIndicator = {};
-            for (var i = 0; i < municipality.length; i++) {
-                tempSviIndicator[municipality[i]] = [];
-            }
-            tempSviSearchElements = Object.keys(weightedCatIndicator);
-            tempSviSearchElements.pop();
-            var sessionKey = Object.keys(weightedCatIndicator[tempSviSearchElements[0]]);
-            sessionKey.pop();
-            sessionKey.pop();
+    for (var p5 = 0; p5 < tempString.length; p5++) {
+        // capture an array for each record
+        var temp;
+        temp = tempString[p5];
+        temp = temp.split("|");
+        generateObject(temp);
+    }
 
-            for (var i = 0; i < tempSviSearchElements.length; i++) {
-                var tempArray = [];
+    Category_PCP_Chart(catData);
 
-                // Grab the municipality of the iteration
-                var tempMunic = weightedCatIndicator[i]["municipality"];
+    /////////////////////////
+    //// Compute the SVI ////
+    /////////////////////////
 
-                // Remove the municipality from the sessionkey array
-                var index = sessionKey.indexOf("municipality");
+    var SVI = {};
+    var sviNameLookUp = 'SVI';
+    var sviJSONthemes = socialVulnIndex;
+    // SVI is an object with region and value
+    SVI = combineIndicators(sviNameLookUp, catData, sviJSONthemes );
 
-                if (index > -1) {
-                    sessionKey.splice(index, 1);
-                }
+    ///////////////
```

Comment:
Now that you have better factorized the scaling, you can remove these lines from here